src/poetry/puzzle/provider.py

```diff
@@ -457,7 +462,16 @@ def incompatibilities_for(
             for dep in dependencies
             if dep.name not in self.UNSAFE_PACKAGES
             and self._python_constraint.allows_any(dep.python_constraint)
-            and (not self._env or dep.marker.validate(self._env.marker_env))
+            and (
+                not self._env
+                or dep.marker.validate(
+                    self._marker_values(
+                        self._active_root_extras
+                        if dependency_package.package.is_root()
+                        else dependency_package.dependency.extras
+                    )
+                )
+            )
```

Comment:
It looks like this change is not required by any test. To be honest, I have currently no idea if this change is redundant or if a test is missing...

Reply:
Ah yeah I wasn't sure what the best policy was here: I just updated the places where I saw markers being evaluated to include extras. What I found is that of the three changes in `provider.py`, any one of them is sufficient to pass the first round of tests. So I think there is some redundancy, but my thought is to not eliminate anything yet due to the issue with uncovered around not resolving conflicts with their own different extras.

Reply:
Confirmed that this change is redundant: either this change or the one on lines 558 are sufficient. Happy to leave this in or to remove it since we haven't found a case depending on this case alone.

In the one or two cases I looked at the other change alone took the same number of steps as with this change, while this change alone took more steps. But I'm not sure if there are some cases I didn't find on a glance where having both changes may reduce the number of steps required.

Reply:
Thank you for digging in and explaining here, that makes sense to me! Reverted